core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -503,21 +503,31 @@ class SnappyConf(@transient val session: SnappySession)
     case Property.PlanCaching.name =>
       value match {
         case Some(boolval) => {
-          if (boolval.toString.toBoolean) session.clearPlanCache()
+          if (boolval.toString.toBoolean) {
+            session.clearPlanCache()
+          }
+          session.planCaching = boolval.toString.toBoolean
         }
         case None =>
       }
 
     case Property.PlanCachingAll.name =>
       value match {
         case Some(boolval) => {
-          val disable_tokenization = !boolval.toString.toBoolean
-          System.setProperty("DISABLE_TOKENIZATION", disable_tokenization.toString)
-          if (disable_tokenization) SnappySession.getPlanCache.asMap().clear()
+          val clear_cache = !boolval.toString.toBoolean
+          if (clear_cache) SnappySession.getPlanCache.asMap().clear()
```

Comment:
current session value should be set: session.planCaching = !clear_cache
also naming convention is camel case "clearCache"

Reply:
Does not matter as I am using this session to disable across all sessions. If I do selectively in this session ( like planCaching above ) then this session in some sense is getting differently treated than others.